pkg/kubelet/config/file_linux_test.go

```diff
@@ -314,22 +319,23 @@ func watchFileChanged(watchDir bool, symlink bool, t *testing.T) {
 			}()
 
 			if watchDir {
-				NewSourceFile(dirName, hostname, 100*time.Millisecond, ch)
+				NewSourceFile(dirName, hostname, period, ch)
 			} else {
-				NewSourceFile(file, hostname, 100*time.Millisecond, ch)
+				NewSourceFile(file, hostname, period, ch)
 			}
+
+			// await fsnotify to be ready
+			time.Sleep(time.Second)
```

Comment:
expectUpdate already waits up to 5 seconds for an event from the channel, not sure why an extra sleep here is needed

Reply:
If `changeFile` is executed before fsnotify is ready, then there will be no fsnotify event.

Because `NewSourcefile` can't guarantee fsnotify to be ready. (the goroutine can be executed later)
https://github.com/kubernetes/kubernetes/blob/0ed8ee6fd7169219fbeb9bca83023a980a7a36a5/pkg/kubelet/config/file_linux.go#L50-L66